share/functions/edit_command_buffer.fish

```diff
@@ -0,0 +1,45 @@
+function edit_command_buffer --description 'Edit the command buffer in an external editor'
+    set -l f (mktemp)
+    if set -q f[1]
+        mv $f $f.fish
+        set f $f.fish
+    else
+        # We should never execute this block but better to be paranoid.
+        if set -q TMPDIR
+            set f $TMPDIR/fish.(echo %self).fish
+        else
+            set f /tmp/fish.(echo %self).fish
+        end
+        touch $f
+        or return 1
+    end
+
+    # Edit the command line with the users preferred editor or vim or emacs.
+    commandline -b >$f
+    if set -q EDITOR
+        eval $EDITOR $f
+    else if command -s vim >/dev/null
+        vim $f
+    else if command -s emacs >/dev/null
+        emacs $f
```

Comment:
Also, since 121109ee4d2ce9f23a6dfe4e377800c75566de85, we have `command -sq`, so this could just be `command -sq emacs`.